pkg/providers/vsphere/config/template-md.yaml

```diff
@@ -83,6 +83,11 @@ spec:
             [plugins."io.containerd.grpc.v1.cri".registry.configs."{{.registryMirrorConfiguration}}".tls]
               ca_file = "/etc/containerd/certs.d/{{.registryMirrorConfiguration}}/ca.crt"
             {{- end }}
+            {{- if .registryAuth }}
+            [plugins."io.containerd.grpc.v1.cri".registry.configs."{{.registryMirrorConfiguration}}".auth]
+              username = "{{.registryUsername}}"
+              password = "{{.registryPassword}}"
```

Comment:
Similar to my other comment on quotes, we'll need additional test case(s).

Reply:
https://github.com/aws/eks-anywhere/pull/3684#discussion_r997547774